File: ch2_labs/lab1_C++/ConsoleApp1/MyString.cpp
#include "MyString.h"
#include <algorithm>

// Конструктор
MyString::MyString(const std::string& input) : str(input) {}

// Перетворення рядка у верхній регістр
std::string MyString::toUpperCase() const {
    std::string upperStr = str;
    std::transform(upperStr.begin(), upperStr.end(), upperStr.begin(), ::toupper);
    return upperStr;
}

// Обчислення довжини рядка
size_t MyString::length() const {
    return str.length();
}

// Заміна символів
void MyString::replaceChar(char oldChar, char newChar) {
    std::replace(str.begin(), str.end(), oldChar, newChar);
}

// Повернення значення рядка
std::string MyString::getValue() const {
    return str;
}

File: ch2_labs/lab1_C#/ConsoleApp1/MyString.cs
using System;

public class MyString
{
    private string str;

    // Конструктор
    public MyString(string input)
    {
        str = input;
    }

    // Метод повертає рядок у верхньому регістрі
    public string ToUpperCase()
    {
        return str.ToUpper();
    }

    // Метод повертає довжину рядка
    public int Length()
    {
        return str.Length;
    }

    // Метод замінює всі входження oldChar на newChar
    public void ReplaceChar(char oldChar, char newChar)
    {
        str = str.Replace(oldChar, newChar);
    }

    // Метод для отримання значення рядка
    public string GetValue()
    {
        return str;
    }
}

